app/models/views/application_result.rb

```diff
@@ -0,0 +1,48 @@
+module Views
+  class ApplicationResult
+
+    def initialize(application)
+      @application = application
+    end
+
+    def result
+      %w[full part none].include?(outcome) ? outcome : 'error'
+    end
+
+    def amount_to_pay
+      if evidence_or_application.amount_to_pay.present?
+        "£#{evidence_or_application.amount_to_pay.round}"
+      end
+    end
+
+    def savings
+      result_translations[@application.savings_investment_valid?.to_s]
+    end
+
+    def income
+      result_translations[%w[full part].include?(result).to_s]
+    end
+
+    private
+
+    def outcome
+      case evidence_or_application
+      when EvidenceCheck
+        evidence_or_application.outcome
+      when Application
+        evidence_or_application.application_outcome
+      end
+    end
```

Comment:
No objections from me. Just wanted to know the thought process behind it. Thanks :+1: